### PR TITLE
Feat: link checking for phishing detection

### PR DESCRIPTION
**Pull request by hamza221**
part of [#9453](https://github.com/nextcloud/mail/issues/9453)
| 1| 2 |
|--------|--------|
| image | image |

### CODE REVIEW

**Review by miaulalala — APPROVED** [review]

File: lib/Service/PhishingDetection/LinkCheck.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
...|sx|sy|sz|tc|td|tf|tg|th|tj|tk|tl|tm|tn|to|tp|tr|tt|tv|tw|tz|ua|ug|uk|us|uy|uz|va|vc|ve|vg|vi|vn|vu|wf|ws|ye|yt|yu|za|zm|zw)\/)(?:[^\s()<>{}\[\]]+|\([^\s()]*?\([^\s()]+\)[^\s()]*?\)|\([^\s]+?\))+(?:\([^\s()]*?\([^\s()]+\)[^\s()]*?\)|\([^\s]+?\)|[^\s`!()\[\]{};:\'".,<>?«»“”‘’])|(?:(?<!@)[a-z0-9]+(?:[.\-][a-z0-9]+)*[.](?:com|net|org|edu|gov|mil|aero|asia|biz|cat|coop|info|int|jobs|mobi|museum|name|post|pro|tel|travel|xxx|ac|ad|ae|af|ag|ai|al|am|an|ao|aq|ar|as|at|au|aw|ax|az|ba|bb|bd|be|bf|bg|bh|bi|bj|bm|bn|bo|br|bs|bt|bv|bw|by|bz|ca|cc|cd|cf|cg|ch|ci|ck|cl|cm|cn|co|cr|cs|cu|cv|cx|cy|cz|dd|de|dj|dk|dm|do|dz|ec|ee|eg|eh|er|es|et|eu|fi|fj|fk|fm|fo|fr|ga|gb|gd|ge|gf|gg|gh|gi|gl|gm|gn|gp|gq|gr|gs|gt|gu|gw|gy|hk|hm|hn|hr|ht|hu|id|ie|il|im|in|io|iq|ir|is|it|je|jm|jo|jp|ke|kg|kh|ki|km|kn|kp|kr|kw|ky|kz|la|lb|lc|li|lk|lr|ls|lt|lu|lv|ly|ma|mc|md|me|mg|mh|mk|ml|mm|mn|mo|mp|mq|mr|ms|mt|mu|mv|mw|mx|my|mz|na|nc|ne|nf|ng|ni|nl|no|np|nr|nu|nz|om|pa|pe|pf|pg|ph|pk|pl|pm|pn|pr|ps|pt|pw|py|qa|re|ro|rs|ru|rw|sa|sb|sc|sd|se|sg|sh|si|sj|Ja|sk|sl|sm|sn|so|sr|ss|st|su|sv|sx|sy|sz|tc|td|tf|tg|th|tj|tk|tl|tm|tn|to|tp|tr|tt|tv|tw|tz|ua|ug|uk|us|uy|uz|va|vc|ve|vg|vi|vn|vu|wf|ws|ye|yt|yu|za|zm|zw)\b\/?(?!@)))/';
```
[the remark this answers is not in the document]
Comment (hamza221) [reply lib/Service/PhishingDetection/LinkCheck.php:25]: https://gist.github.com/gruber/8891611

[the remark this answers is not in the document]
Comment (hamza221) [reply lib/Service/PhishingDetection/LinkCheck.php:25]: https://dev.horde.org/api/master/lib/Text_Filter/classes/Horde_Text_Filter_Linkurls.html  is based on it

[the remark this answers is not in the document]
Comment (ChristophWurst) [reply lib/Service/PhishingDetection/LinkCheck.php:25]: Yes, link to the source pls

**Review by ChristophWurst — APPROVED** [review]
Looks good

I'd really love to see a bit of test coverage for the new check

I'm okay with merging without to meet the branch-off and backport the tests in the near future

**Pull request merged**